xbmc/interfaces/legacy/ListItem.cpp

```diff
@@ -482,6 +482,8 @@ namespace XBMCAddon
             videotag.SetTrailer(value);
           else if (key == "path")
             videotag.SetPath(value);
+          else if (key == "filenameandpath")
```

Comment:
its to fix a problem with addons not being able to set the filenameandpath of the videotag.
this in turn fixes some handeling of items when working out if they are folders or not later on when an item is played from the info screen.